base/deltas_ee.go

```diff
@@ -26,3 +26,8 @@ func Diff(old, new map[string]interface{}) (delta []byte, err error) {
 func Patch(old *map[string]interface{}, delta []byte) (err error) {
 	return fleecedelta.PatchJSON(old, delta)
 }
+
+// PatchJSONWithMap attempts to path old with the given delta passed as a map[string]interface{}
+func PatchMap(old *map[string]interface{}, delta map[string]interface{}) (err error) {
```

Comment:
My vote is to replace the existing Patch function the one that takes `interface{}`. There's only one other instance of `Patch()` being used (in the BLIP test client), and that's doing the same marshal -> unmarshal work that we're trying to remove.

Let's try and keep the SG side of this minimal, as we're not concerned about other types of Patching here.